File: TransCoder_tokenized_test_set_functions/java/LONGEST_SUBSEQUENCE_DIFFERENCE_ADJACENTS_ONE_SET_2.java
static int longLenSub ( int [ ] arr, int n ) {
  HashMap < Integer, Integer > um = new HashMap < Integer, Integer > ( );
  int longLen = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    int len = 0;
    if ( um . containsKey ( arr [ i ] - 1 ) && len < um . get ( arr [ i ] - 1 ) ) len = um . get ( arr [ i ] - 1 );
    if ( um . containsKey ( arr [ i ] + 1 ) && len < um . get ( arr [ i ] + 1 ) ) len = um . get ( arr [ i ] + 1 );
    um . put ( arr [ i ], len + 1 );
    if ( longLen < um . get ( arr [ i ] ) ) longLen = um . get ( arr [ i ] );
  }
  return longLen;
}

File: TransCoder_tokenized_test_set_functions/python/LONGEST_SUBSEQUENCE_DIFFERENCE_ADJACENTS_ONE_SET_2.py
def longLenSub ( arr, n ) :
    um = defaultdict ( lambda : 0 )
    longLen = 0
    for i in range ( n ) :
        len1 = 0
        if ( arr [ i - 1 ] in um and len1 < um [ arr [ i ] - 1 ] ) :
            len1 = um [ arr [ i ] - 1 ]
        if ( arr [ i ] + 1 in um and len1 < um [ arr [ i ] + 1 ] ) :
            len1 = um [ arr [ i ] + 1 ]
        um [ arr [ i ] ] = len1 + 1
        if longLen < um [ arr [ i ] ] :
            longLen = um [ arr [ i ] ]
    return longLen
